pages/index.js

```diff
@@ -17,11 +17,57 @@ import { load, loadAuthFields } from 'ducks/articleList';
 import { mainStyle } from './index.styles';
 
 class Index extends ListPage {
+  state = {
+    localEditorHelperList: {
+      demoId: {
+        // ID of articles state which already read or replied
+        read: true,
+        notArticleReplied: false, // false ||
+      },
+    },
+  };
+
   componentDidMount() {
     // Browser-only
     this.props.dispatch(loadAuthFields(this.props.query));
+    this.initLocalEditorHelperList();
   }
 
+  initLocalEditorHelperList = () => {
+    if (localStorage) {
+      const localEditorHelperList = JSON.parse(
+        localStorage.getItem('localEditorHelperList')
+      );
+      localEditorHelperList &&
+        this.setState({
+          localEditorHelperList,
+        });
+    }
+  };
+
+  handleLocalEditorHelperList = ({ id, read, notArticleReplied }) => {
+    this.setState(
```

Comment:
Since we are reading and setting `localEditorHelperList` at the same time, we should use the updater function style instead:

```
this.setState(({localEditorHelperList}) => ({
  ...localEditorHelperList,
  [id]: {read, notArticleReplied}
}), cb);
```

Ref: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous